storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -201,6 +205,7 @@ public Optional<String> serverConfigName(String configName) {
     // Visible for testing
     public static final Set<String> CONFIGS_WITH_NO_SERVER_DEFAULTS = Collections.unmodifiableSet(Utils.mkSet(
             TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG,
+            TopicConfig.REMOTE_LOG_DISABLE_POLICY_CONFIG,
```

Comment:
I see retain is set as default, so do we need this ?

Reply:
I believe this is needed because while we have a default on the topic-level we don't have a default on the server-level. I follow the same logic as the `remote.storage.enable` configuration